zebrad/src/components/tokio.rs

```diff
@@ -23,3 +25,87 @@ impl TokioComponent {
         })
     }
 }
+
+/// Zebrad's graceful shutdown function, blocks until one of the supported
+/// shutdown signals is received.
+async fn shutdown() {
+    imp::shutdown().await;
+}
+
+/// Extension trait to centralize entry point for runnable subcommands that
+/// depend on tokio
+pub(crate) trait RuntimeRun {
+    fn run(&mut self, fut: impl Future<Output = Result<(), Report>>);
+}
+
+impl RuntimeRun for Runtime {
+    fn run(&mut self, fut: impl Future<Output = Result<(), Report>>) {
+        let result = self.block_on(async move {
+            tokio::select! {
+                result = fut => result,
+                _ = shutdown() => Ok(()),
+            }
+        });
+
+        match result {
+            Ok(()) => {}
+            Err(e) => {
+                eprintln!("Error: {:?}", e);
+                app_writer().shutdown(Shutdown::Forced);
+            }
+        }
+    }
+}
+
+#[cfg(unix)]
+mod imp {
+    use tokio::signal::unix::{signal, SignalKind};
+    use tracing::info;
+
+    pub(super) async fn shutdown() {
+        tokio::select! {
+            // SIGINT  - To allow Ctrl-c to emulate SIGTERM while developing.
+            () = sig(SignalKind::interrupt(), "SIGINT") => {}
+            // SIGTERM - Kubernetes sends this to start a graceful shutdown.
```

Comment:
This isn't actually a Kubernetes-specific signal.

```suggestion
            // SIGTERM - Standard shutdown signal used by process launchers.
```